esp-wifi/src/lib.rs

```diff
@@ -117,97 +117,116 @@ fn init_heap() {
 }
 
 #[cfg(any(feature = "esp32c3", feature = "esp32c2", feature = "esp32c6"))]
-/// Initialize for using WiFi / BLE
-/// This will initialize internals and also initialize WiFi and BLE
-pub fn initialize(
-    systimer: Alarm<Target, 0>,
-    rng: hal::Rng,
-    radio_clocks: hal::system::RadioClockControl,
-    clocks: &Clocks,
-) -> Result<(), InitializationError> {
-    init_heap();
+type Timer = Alarm<Target, 0>;
 
-    #[cfg(feature = "esp32c6")]
-    if clocks.cpu_clock != MegahertzU32::MHz(160) {
-        return Err(InitializationError::WrongClockConfig);
-    }
+#[cfg(any(feature = "esp32", feature = "esp32s3", feature = "esp32s2"))]
+type Timer = hal::timer::Timer<hal::timer::Timer0<hal::peripherals::TIMG1>>;
```

Comment:
Should we give those types a public visibility, so that if someone were to wrap `initialize()` inside another function, they wouldn't have to redefine those types again.

Reply:
You mean just for convenience? 🤔 Sure that shouldn't be a problem. Not sure if the naming is a bit too generic then - probably better to rename this if we make it public